admin-console/src/main/java/com/agiletec/apsadmin/portal/PageTreeAction.java

```diff
@@ -19,183 +19,266 @@
+            _logger.error("error in movePage", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    public String moveTree() {
+        String selectedNode = this.getSelectedNode();
+        String parentPageCode = this.getRequest().getParameter("parentPageCode");
+        if (null == parentPageCode) {
+            parentPageCode = this.getPageManager().getRoot().getCode();
+        }
+        try {
+            String check = this.checkMovePage(selectedNode, parentPageCode);
+            if (null != check) {
+                return check;
+            }
+            IPage currentPage = this.getPage(selectedNode);
+            IPage parent = this.getPage(parentPageCode);
+            this.getPageManager().movePage(currentPage, parent);
+        } catch (Throwable t) {
+            _logger.error("error in move page", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    protected String checkMovePage(String selectedNode, String parentPageCode) {
+        IPage currentPage = this.getPage(selectedNode);
+        if (null == currentPage) {
+            _logger.info("Required a selected node");
+            this.addActionError(this.getText("error.page.selectPage"));
+            return "pageTree";
+        }
+        if (currentPage.getCode().equals(this.getPageManager().getRoot().getCode())) {
+            _logger.info("Root page cannot be moved");
+            this.addActionError(this.getText("error.page.move.rootNotAllowed"));
+            return "pageTree";
+        }
+        if ("".equals(parentPageCode) || null == this.getPageManager().getPage(parentPageCode)) {
+            this.addActionError(this.getText("error.page.move.selectPageParent"));
+            return "pageTree";
+        }
+        IPage parent = this.getPage(parentPageCode);
+        if (null == parent) {
+            _logger.info("Required a selected node");
+            this.addActionError(this.getText("error.page.selectPageParent"));
+            return "pageTree";
+        }
+        if (parent.getCode().equals(currentPage.getParentCode())) {
+            _logger.debug("trying to move a node under it's own parent..");
+            this.addActionError(this.getText("error.page.move.parentUnderChild.notAllowed"));
+            return "pageTree";
+        }
+        if (parent.isChildOf(selectedNode)) {
+            List<String> args = new ArrayList<String>();
+            args.add(parent.getCode());
+            args.add(selectedNode);
+            this.addActionError(this.getText("error.page.move.parentUnderChild.notAllowed"));
+            return "pageTree";
+        }
+        //group check
+        if (!currentPage.getGroup().equals(Group.FREE_GROUP_NAME)
+                && !currentPage.getGroup().equals(parent.getGroup())) {
+            this.addActionError(this.getText("error.page.move.differentRestriction.notAllowed"));
+            return "pageTree";
+        }
+        return null;

```

Comment:
this is a brand new method that is either returning pageTree if an error occurs or null if it is successful.
Couldn't this just be returning a boolean at this point and have the method consuming this return pageTree?